.github/workflows/ci.yaml

```
@@ -0,0 +1,137 @@
+name: CI
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+env:
+  CARGO_TERM_COLOR: always
+  RUST_BACKTRACE: full
+jobs:
+  pre-commit:
+    name: Validate `pre-commit` Hooks
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout sources
+      uses: actions/checkout@v3
+    - name: Get python
+      uses: actions/setup-python@v3
+    - name: Run pre-commit check
+      uses: pre-commit/action@v3.0.1
+      with:
+        extra_args: run --all-files --hook-stage manual
+  conventional_commits:
+    name: Conventional Commit Checking
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@main
+        with:
+          fetch-depth: 0
+      - name: Assert conventional commit usage
+        uses: oknozor/cocogitto-action@v3
+  spelling:
+    name: Spell Checking
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout sources
+      uses: actions/checkout@v4
+    - name: typos
+      uses: crate-ci/typos@master
+  lychee_links:
+    name: Broken URL Checking
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v4
+      - name: Run lychee
+        uses: lycheeverse/lychee-action@v1.9.3
+        with:
+          fail: true
```

question: why let this fail?

---

is false it will only log warnings and exit with code 0, job won't fail.